### PR TITLE
Clear caches after the Ballerina toml update to get diagnostics

### PR DESCRIPTION
**Pull request by Dilhasha**
Fixes https://github.com/ballerina-platform/ballerina-lang/issues/34378

## Check List 
- [x] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

**Review by malinthar — COMMENTED** [review]
IMO, Clearing the cache every time we update the Ballerina.toml could slow things down. Shall we try to identify if the package name or org name has changed, and clear the caches only in those cases?

**Comment by IMS94** [comment]
> IMO, Clearing the cache every time we update the Ballerina.toml could slow things down. Shall we try to identify if the package name or org name has changed, and clear the caches only in those cases?

+1

**Review by gayaldassanayake — APPROVED** [review]

**Comment by Dilhasha** [comment]
When there's a toml update, we are updating all the modules of the package and create a new package. This replaces the package cache with the newly created package. When there is a  `package` or `org` change, this does not replace the existing package in the cache due to the name change. Instead, it creates a new package. So, since the package with the old name is available in the cache, it doesn't produce a diagnostic. 

With a8730a406e961bb981d11f9eb86fc52e8b790446 , we clear the old package from cache before creating the new package.

**Review by malinthar — DISMISSED** [review]
LGTM!

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/Package.java [hunk 1]
```diff
@@ -697,6 +697,9 @@ private void updateModules() {
                         oldModuleContext.isDefaultModule(), srcDocContextMap, testDocContextMap,
                         oldModuleContext.moduleMdContext().orElse(null),
                         oldModuleContext.moduleDescDependencies(), resourceMap, testResourceMap));
+                // Clear compilation package cache
```
Comment (azinneera) [inline compiler/ballerina-lang/src/main/java/io/ballerina/projects/Package.java:700]:
```suggestion
                // Remove the module with old PackageID from the compilation cache
```

**Review by azinneera — APPROVED** [review]

File: compiler/ballerina-lang/src/main/java/io/ballerina/projects/Package.java (hunk 1 above)
Comment (Dilhasha) [reply compiler/ballerina-lang/src/main/java/io/ballerina/projects/Package.java]: Fixed with f4ad8c6fe9bdff0b316bd3cef1b17b8804dd0a05

**Pull request merged**